tests/unit_tests/test_mutox.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import pytest
+import torch
+from torch import nn
+
+from sonar.models.mutox.builder import (
+    MutoxClassifierBuilder,
+    MutoxConfig,
+    create_mutox_model,
+)
+from sonar.models.mutox.classifier import MutoxClassifier
+from sonar.models.mutox.loader import convert_mutox_checkpoint
+
+# Builder tests
+
+
+@pytest.mark.parametrize("input_size", [256, 512, 1024])
+@pytest.mark.parametrize("device", [torch.device("cpu")])
+@pytest.mark.parametrize("dtype", [torch.float32, torch.float64])
+def test_mutox_classifier_builder(input_size, device, dtype):
+    """Test MutoxClassifierBuilder initializes a model with correct configuration and dtype."""
+    config = MutoxConfig(input_size=input_size)
+    builder = MutoxClassifierBuilder(config, device=device, dtype=dtype)
+    model = builder.build_model()
+
+    # Check if model layers are correctly initialized with shapes
+    assert isinstance(model, nn.Module), "Model should be an instance of nn.Module"
+    assert all(
+        isinstance(layer, nn.Sequential) for layer in model.model_all.children()
+    ), "All layers should be instances of nn.Sequential"
+
+    test_input = torch.zeros((5, input_size), device=device, dtype=dtype)
+    result = model(test_input)
+    assert result.shape == (5, 1), f"Expected output shape (5, 1), got {result.shape}"
+
+
+@pytest.mark.parametrize("input_size", [256, 512])
+def test_create_mutox_model(input_size):
+    """Test create_mutox_model function to confirm it creates a model with the specified config."""
+    config = MutoxConfig(input_size=input_size)
+    model = create_mutox_model(config, device=torch.device("cpu"))
+
+    # Check if the created model has the expected structure and behavior
+    test_input = torch.zeros((3, input_size))
+    result = model(test_input)
+    assert result.shape == (3, 1), f"Expected output shape (3, 1), got {result.shape}"
+    assert isinstance(model, nn.Module), "Model should be an instance of nn.Module"
+
+
+# Classifier tests
+
+
+def test_mutox_classifier_forward():
+    """Test that MutoxClassifier forward pass returns expected output shape."""
+    test_model = nn.Sequential(
+        nn.Linear(10, 5),
+        nn.ReLU(),
+        nn.Linear(5, 1),
+    )
+    model = MutoxClassifier(test_model)
+
+    test_input = torch.randn(3, 10)
+    output = model(test_input)
```

Comment:
Actually, this test is working! It is breaking right now, because the at some point, the matrix shapes are not compatible (as a result of messing with the last linear layer). 

I am not sure if you are paying attention to the failing tests in the PR interface, but they are actually informative and useful:
<img width="903" alt="image" src="https://github.com/user-attachments/assets/d5bbd2cb-1ce8-49bb-9690-3f6e6e6a7325">

Reply:
I have been reading them, I resolved most of the problems related to mypy that appeared after the first migration of the code